src/com/vivian/java/leetcode/CCI2_1RemoveDupsWithoutBuffer.java

```diff
@@ -0,0 +1,54 @@
+package com.vivian.java.leetcode;
+import java.lang.NullPointerException;
+/* write code to remove duplicates from an unsorted linkedList
+(how would you solve this problem if a temporary buffer is not allowed?)
+
+clarification: without buffer
+               Node style linkedlist
+
+approach: use three pointer: cur(iterator for grand loop; the end edge of inner loop)
+                             pre(maker the current non_duplicate element)
+                             run(inner loop, find the duplicate)
+
+complexity: runtime complexity o(n*n), memory complexity o(1) only for the variables
+*/
+
+public class CCI2_1RemoveDupsWithoutBuffer {
+    public static class Node<T> {
```

Comment:
I thought you have moved Node out to it's own file. Can you use that instead?

Reply:
the faction is remove repeated node out its own file

Reply:
Not sure I follow.